torch/_utils.py

```diff
@@ -48,6 +48,38 @@ def _type(self, dtype=None, non_blocking=False, **kwargs):
     return dtype(self.size()).copy_(self, non_blocking)
 
 
+def _hpu(self, device=None, non_blocking=False, **kwargs):
+    """Returns a copy of this object in HPU memory.
+
+    If this object is already in HPU memory and on the correct device, then
+    no copy is performed and the original object is returned.
+
+    Args:
+        device (int): The destination HPU id. Defaults to the current device.
+        non_blocking (bool): If ``True`` and the source is in pinned memory,
+            the copy will be asynchronous with respect to the host. Otherwise,
+            the argument has no effect.
+        **kwargs: For compatibility, may contain the key ``async`` in place of
+            the ``non_blocking`` argument.
+    """
+    non_blocking = _get_async_or_non_blocking("hpu", non_blocking, kwargs)
+    hpu = getattr(torch, "hpu", None)
+    assert hpu is not None, "HPU device module is not loaded"
```

Comment:
is it guaranteed that hpu will be available if hasattr(torch, "hpu")? do we have an equivalent of `torch.{device}.is_available()` for hpu?

Reply:
Actually, availability of  torch.hpu only means that the module has been loaded and subsequent check for actual device can be done with torch.hpu.is_available, just like you suspect. Only that at this level (_utils.py) there are no such checks.

Reply:
hm why is it not available in `_utils.py`? I see that `torch.xpu.is_available()` is used in this file

Reply:
Got it, thanks for clarifying